diff --git a/src/doom/d_main.c b/src/doom/d_main.c
index 2092987d..4804c700 100644
--- a/src/doom/d_main.c
+++ b/src/doom/d_main.c
@@ -79,6 +79,10 @@
 
 #include "d_main.h"
 
+#ifdef DMCP
+#include "dmcp_integration.h"
+#endif
+
 #include "doom_icon.c"
 
 //
@@ -2445,6 +2449,10 @@ void D_DoomMain (void)
 	    D_StartTitle ();                // start up intro loop
     }
 
+#ifdef DMCP
+    DMCP_Init();
+#endif
+
     D_DoomLoop ();  // never returns
 }
 
diff --git a/src/doom/g_game.c b/src/doom/g_game.c
index 2af0fb32..56088b03 100644
--- a/src/doom/g_game.c
+++ b/src/doom/g_game.c
@@ -50,6 +50,10 @@
 
 #include "d_main.h"
 
+#ifdef DMCP
+#include "dmcp_integration.h"
+#endif
+
 #include "wi_stuff.h"
 #include "hu_stuff.h"
 #include "st_stuff.h"
@@ -1657,6 +1661,10 @@ void G_Ticker (void)
 	D_PageTicker (); 
 	break;
     }        
+
+#ifdef DMCP
+    DMCP_Tick();
+#endif
 } 
  
  
